src/core/historymgr.cpp

```diff
@@ -156,16 +160,35 @@ void HistoryMgr::add(const QString &p_path,
     emit historyUpdated();
 }
 
-void HistoryMgr::insertHistoryItem(QVector<HistoryItem> &p_history, const HistoryItem &p_item)
+void HistoryMgr::remove(const QVector<QString> &p_paths, Notebook *p_notebook)
+{
+    foreach(QString p_itemPath, p_paths) {
```

Comment:
Note: Since Qt 5.7, the use of this macro is discouraged. It will be removed in a future version of Qt. Please use C++11 range-for, possibly with qAsConst(), as needed.

Reply:
Done.